core/types/transaction.go

```diff
@@ -272,14 +272,34 @@ func (tx *Transaction) String() string {
 // Transaction slice type for basic sorting.
 type Transactions []*Transaction
 
-func (s Transactions) Len() int      { return len(s) }
+// Len returns the length of s
+func (s Transactions) Len() int { return len(s) }
+
+// Swap swaps the i'th and the j'th element in s
 func (s Transactions) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 
+// GetRlp implements Rlpable and returns the i'th element of s in rlp
 func (s Transactions) GetRlp(i int) []byte {
 	enc, _ := rlp.EncodeToBytes(s[i])
 	return enc
 }
 
+// Returns a new set t which is the difference between a to b
+func TxDifference(a, b Transactions) (t Transactions) {
+	m := make(map[common.Hash]*Transaction)
+	for _, tx := range b {
+		m[tx.Hash()] = tx
+	}
+
+	for _, tx := range a {
+		if _, ok := m[tx.Hash()]; !ok {
+			t = append(t, tx)
+		}
+	}
```

Comment:
Also maybe rename `m` to `remove` and `t` to `keep`. It's not really nice to use one letter variables, especially since they seem quire arbitrary to me.